lib/actions/index.js

```diff
@@ -1,6 +1,38 @@
-export const createTrail = (payload) => {
-  return {
-    type: 'CREATE_TRAIL',
-    payload
+import firebase from 'firebase';
+
+const firebaseTrails = firebase.database().ref('trails');
+
+export const fetchAllTrails = () => {
+  return (dispatch, getState) => {
+    let fetchedTrails = [];
+
+    firebaseTrails.once('value').then(result => {
+      result.forEach(trail => {
+        fetchedTrails.push(trail.val());
+      });
+
+      dispatch({
+        type: 'RECEIVE_ALL_TRAILS',
+        trails: fetchedTrails
+      });
+    });
+  };
+};
+
+export const createTrail = (trailData) => {
+  return (dispatch) => {
+
+    let newTrailKey = firebaseTrails.push().key;
+
+      firebaseTrails.child(newTrailKey).set(trailData)
+      .then(() => {
+        dispatch({
+          type: 'CREATE_NEW_TRAIL',
+          trail: trailData
```

Comment:
Verify that `trailData` is an object that contains the data you want sent to your reducer. Even if it's just adding a `console.log` before the `dispatch` call.